src/snowflake/snowpark/async_job.py

```diff
@@ -355,6 +357,25 @@ def result(
                 )
 
         if async_result_type == _AsyncResultType.NO_RESULT:
+            # The following section is copied from python connector.
+            # Later we should expose it from python connector and reuse it.
+            retry_pattern_pos = 0
+            while True:
```

Comment:
would maybe use a timeout / retry count instead of endless loop. This looks like an antipattern to me.

Reply:
This is to polling a query result so we may not want to timeout from the client. It's not like a connection.